Tasks/ServiceFabricUpdateAppVersions/Find-FileChanges.psm1

```diff
@@ -0,0 +1,114 @@
+﻿function Get-PackageFiles ($Path)
+{
+    # List all files except '.pdb' files because their content changes every build even with the 'deterministic' compiler flag
+    Find-VstsFiles -LiteralDirectory $Path -LegacyPattern "**;-:**\*.pdb" -Force |
+        ForEach-Object { $_.ToLower() } |
+        Sort-Object
+}
+
+function Find-FileChanges
+{
+    [CmdletBinding()]
+    [OutputType([bool])]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path $_})]
+        [string]
+        $NewPackageRoot,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path $_})]
+        [string]
+        $OldPackageRoot,
+
+        [string]
+        $LogIndent,
+
+        [Switch]
+        $LogAllChanges
+    )
+
+    $LogIndent += "".PadLeft(2)
+
+    $newFilesQueue = [System.Collections.Queue] @(Get-PackageFiles $NewPackageRoot)
+    $oldFilesQueue = [System.Collections.Queue] @(Get-PackageFiles $OldPackageRoot)
+
+    $result = $false
+
+    # Search for changes while both queues have at least one file
+    while ($oldFilesQueue.Count -gt 0 -and $newFilesQueue.Count -gt 0)
+    {
+        $newFile = $newFilesQueue.Peek()
+        $oldFile = $oldFilesQueue.Peek()
+
+        $newRelativePath = $newFile.Substring($NewPackageRoot.Length + 1)
+        $oldRelativePath = $oldFile.Substring($OldPackageRoot.Length + 1)
+
+        if ($oldRelativePath -lt $newRelativePath)
+        {
+            Write-Host "$LogIndent$(Get-VstsLocString -Key FileRemoved -ArgumentList $oldRelativePath)"
+            $result = $true
+            $oldFilesQueue.Dequeue() | Out-Null
+        }
+        elseif ($oldRelativePath -gt $newRelativePath)
+        {
+            Write-Host "$LogIndent$(Get-VstsLocString -Key FileAdded -ArgumentList $newRelativePath)"
+            $result = $true
+            $newFilesQueue.Dequeue() | Out-Null
+        }
+        else
+        {
+            # Based on my testing, using 'Get-FileHash' and '-ne' is about:
+            # 50x faster than using 'Get-Content', 'Out-String', and '-ne'
+            # 10-15x faster than using 'Get-Content' and 'Compare-Object'
+            $newHash = Get-FileHash $newFile -Algorithm SHA256
+            $oldHash = Get-FileHash $oldFile -Algorithm SHA256
```

Comment:
Get-FileHash isn't available until PowerShell 4.0, so I removed that usage. I also refactored to use a new module 'Test-FileEqual' so that unit tests don't have to know my implementation details of file equality